### DISCUSSION TITLE
Questions about “statdose”

### DISCUSSION THREAD

**Discussion by CQUswl** [nrc-cnrc/EGSnrc#1088] [General]
When reading volume modelled dose with “statdose”, you can only read the dose in the direction parallel to the X-axis or Y-axis, how can you read the dose distribution on the diagonal?

**Comment by crcrewso** [comment]
That's a really good question. Last time I needed diagonals I reran the dosxyznrc simulation rotating the source 45 degrees.

**Comment by CQUswl** [comment]
> That's a really good question. Last time I needed diagonals I reran the dosxyznrc simulation rotating the source 45 degrees.
This is not a bad solution. But re-running the simulation wastes a lot of time. I think there must be another way to solve it.

**Reply by rtownson** [reply]
There's not a tool to do this within EGSnrc. You could write your own script. You'd just need to do 2D interpolations between the 4 surrounding voxels, assuming the profile is still in line with the 3rd axis, for an arbitrary angle. If the pixels are square and you only want the 45 degree angle, just read in the 3ddose file and pick out the voxels on the diagonal of interest (after each row, increment the column number by 1). In python I imagine you could do this in a few dozen lines or less, if you have some programming skills.

**Reply by CQUswl** [reply]
I understand,Thanks.
